solidity/contracts/hooks/OPL2ToL1CcipReadHook.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+// ============ Internal Imports ============
+import {TokenMessage} from "../token/libs/TokenMessage.sol";
+import {StandardHookMetadata} from "./libs/StandardHookMetadata.sol";
+import {AbstractPostDispatchHook} from "./libs/AbstractPostDispatchHook.sol";
+import {AbstractMessageIdAuthHook} from "./libs/AbstractMessageIdAuthHook.sol";
+import {InterchainGasPaymaster} from "../hooks/igp/InterchainGasPaymaster.sol";
+
+/**
+ * @title OPL2ToL1CcipReadHook
+ * @notice Inform an OPL2ToL1ProveWithdrawalIsm that a withdrawal has been executed
+ * on L2
+ * @dev We expect a CCIP-read ISM executing portal.proveWithdrawal() on destination
+ * after 7 days
+ */
+contract OPL2ToL1CcipReadHook is AbstractMessageIdAuthHook {
+    using StandardHookMetadata for bytes;
+    using TokenMessage for bytes;
+
+    // ============ Constants  ============
+    uint32 public constant MIN_GAS_LIMIT = 500_000;
+
+    // ============ Constructor ============
+    constructor(
+        address _mailbox,
+        uint32 _destinationDomain,
+        bytes32 _ism
+    ) AbstractMessageIdAuthHook(_mailbox, _destinationDomain, _ism) {}
+
+    /// @inheritdoc AbstractPostDispatchHook
+    function _quoteDispatch(
+        bytes calldata metadata,
+        bytes calldata message
+    ) internal view override returns (uint256) {
+        return mailbox.quoteDispatch(destinationDomain, ism, message, metadata);
```

Comment:
IIUC, this is for the quote in the sendMessageId dispatch yeah? If so, imo the metadata should include the gasLimit that the `OPL2ToL1ProveWithdrawalIsm` needs for the proving of the withdrawal, no?

Reply:
Yes, that's it. I did it but it looked like it didn't change anything. For instance if I override the StandardHookMetadata gas limit into the `sendMessageId` dispatch, I still see just one single payment to the IGP, is that correct? Shouldn't be there two payment events?